### ISSUE TITLE
Better auto-join behaviour for group chats

### ISSUE THREAD

**Issue by fancypantalons** [hoehermann/purple-signald#13]
Figured I'd break this into its own issue.

Right now, if a user is a member of a group chat, the plugin automatically opens a window in Pidgin for that chat at startup.  And this is definitely pretty annoying.

The goal is to make this behaviour more flexible.

In particular, I'm proposing a configuration setting for the account that controls auto-join behaviour, with three settings:

1. Never automatically open
2. Automatically open when a message is received or when invited to a group chat
3. Always open

Of course, if the user has selected 1 or 2, they'll need some way to open existing group chats, so the proposal is to also automatically add group chats to the buddy list (with the `gtk-persistent` flag set by default), which will allow the user to then open the chats on demand.

**Comment by fancypantalons** [comment]
Just FYI, I've started a new branch to do this work, and as a starting point I've pushed 5a9920046908808f746c750e9408bc40abaadd93

This is a significant refactoring of the code, which will act as the basis for implementing this concept.

This code has *not* been very thoroughly tested yet.  I've only done some basic smoke testing to validate it doesn't completed break things.  That said, a pair of eyes on it certainly can't hurt...
